Drop Vector constructor taking a raw pointer and a size

cdumez opened the pull request:
#### 7263d3ab568094eb94dd4e9ec859d14a9cd871b0
<pre>
Drop Vector constructor taking a raw pointer and a size
<a href="https://bugs.webkit.org/show_bug.cgi?id=270944">https://bugs.webkit.org/show_bug.cgi?id=270944</a>

Reviewed by Sihui Liu, Mike Wyrzykowski, Justin Michaud and Darin Adler.

Drop Vector constructor taking a raw pointer and a size and port call sites to
the constructor taking a std::span instead.

This is work towards more widespread std::span adoption in WebKit, which has
security benefits.

* Source/JavaScriptCore/runtime/ArrayBuffer.cpp:
(JSC::ArrayBuffer::tryCreate):
* Source/JavaScriptCore/runtime/ArrayBuffer.h:
* Source/JavaScriptCore/runtime/ArrayBufferView.h:
(JSC::ArrayBufferView::bytes const):
* Source/JavaScriptCore/runtime/ISO8601.cpp:
(JSC::ISO8601::parseTimeZoneBracketedAnnotation):
(JSC::ISO8601::parseCalendar):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::formatRange):
(JSC::IntlDateTimeFormat::formatRangeToParts):
* Source/JavaScriptCore/wasm/WasmParser.h:
(JSC::Wasm::Parser::sourceSpan const):
* Source/JavaScriptCore/wasm/WasmSectionParser.cpp:
(JSC::Wasm::SectionParser::parseInitExpr):
* Source/JavaScriptCore/wasm/WasmStreamingParser.cpp:
(JSC::Wasm::StreamingParser::consume):
* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
(PropertyData.convertStringToCppFormat):
* Source/WTF/wtf/Algorithms.h:
(WTF::equalSpans):
* Source/WTF/wtf/Vector.h:
(WTF::Vector::Vector):
(WTF::Vector::subVector const):
* Source/WTF/wtf/cf/VectorCF.h:
(WTF::vectorFromCFData):
* Source/WTF/wtf/cocoa/VectorCocoa.h:
(WTF::vectorFromNSData):
* Source/WTF/wtf/persistence/PersistentEncoder.h:
(WTF::Persistence::Encoder::bytes const):
* Source/WTF/wtf/text/IntegerToStringConversion.h:
* Source/WTF/wtf/text/StringParsingBuffer.h:
* Source/WTF/wtf/text/WTFString.cpp:
(asciiDebug):
* Source/WTF/wtf/text/WTFString.h:
(WTF::String::fromUTF8):
* Source/WebCore/Modules/ShapeDetection/Implementation/Cocoa/FaceDetectorImplementation.mm:
(WebCore::ShapeDetection::convertLandmark):
* Source/WebCore/Modules/async-clipboard/ClipboardItem.cpp:
(WebCore::ClipboardItem::blobFromString):
* Source/WebCore/Modules/async-clipboard/ios/ClipboardImageReaderIOS.mm:
(WebCore::ClipboardImageReader::readBuffer):
* Source/WebCore/Modules/async-clipboard/mac/ClipboardImageReaderMac.mm:
(WebCore::ClipboardImageReader::readBuffer):
* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::consumeArrayBuffer):
(WebCore::FetchBody::consumeArrayBufferView):
(WebCore::FetchBody::consumeText):
(WebCore::FetchBody::bodyAsFormData const):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::FetchBodyConsumer::packageFormData):
(WebCore::resolveWithTypeAndData):
(WebCore::FetchBodyConsumer::resolveWithData):
(WebCore::FetchBodyConsumer::resolveWithFormData):
(WebCore::FetchBodyConsumer::resolve):
(WebCore::FetchBodyConsumer::takeAsText):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.h:
* Source/WebCore/Modules/fetch/FetchBodyOwner.cpp:
(WebCore::FetchBodyOwner::arrayBuffer):
(WebCore::FetchBodyOwner::formData):
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::RTCDataChannel::didReceiveRawData):
* Source/WebCore/Modules/notifications/NotificationDataCocoa.mm:
(WebCore::NotificationData::fromDictionary):
* Source/WebCore/Modules/push-api/PushEvent.cpp:
(WebCore::dataFromPushMessageDataInit):
* Source/WebCore/Modules/push-api/PushManager.cpp:
(WebCore::PushManager::subscribe):
* Source/WebCore/Modules/push-api/PushMessageCrypto.cpp:
(WebCore::PushCrypto::ClientKeys::generate):
(WebCore::PushCrypto::decryptAESGCMPayload):
* Source/WebCore/Modules/push-api/PushMessageData.cpp:
(WebCore::PushMessageData::text):
* Source/WebCore/Modules/push-api/cocoa/PushCryptoCocoa.cpp:
(WebCore::PushCrypto::P256DHKeyPair::generate):
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp:
(WebCore::convertBytesToVector):
* Source/WebCore/Modules/webauthn/cbor/CBORValue.cpp:
(cbor::CBORValue::CBORValue):
* Source/WebCore/Modules/webauthn/fido/DeviceResponseConverter.cpp:
(fido::decodeResponseMap):
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.cpp:
(fido::FidoHidInitPacket::createFromSerializedData):
(fido::FidoHidContinuationPacket::createFromSerializedData):
* Source/WebCore/Modules/webauthn/fido/Pin.cpp:
(fido::pin::encodeCOSEPublicKey):
* Source/WebCore/Modules/webauthn/fido/U2fResponseConverter.cpp:
(fido::WebCore::extractECPublicKeyFromU2fRegistrationResponse):
(fido::WebCore::extractCredentialIdFromU2fRegistrationResponse):
(fido::WebCore::createFidoAttestationStatementFromU2fRegisterResponse):
(fido::readU2fSignResponse):
* Source/WebCore/PAL/pal/text/TextCodec.cpp:
(PAL::TextCodec::decode):
* Source/WebCore/PAL/pal/text/TextCodec.h:
* Source/WebCore/PAL/pal/text/TextEncodingDetector.h:
* Source/WebCore/PAL/pal/text/TextEncodingDetectorICU.cpp:
(PAL::detectTextEncoding):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
(WebCore::fulfillPromiseWithArrayBuffer):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::toKeyData):
(WebCore::copyToVector):
(WebCore::SubtleCrypto::encrypt):
(WebCore::SubtleCrypto::decrypt):
(WebCore::SubtleCrypto::sign):
(WebCore::SubtleCrypto::digest):
(WebCore::SubtleCrypto::deriveBits):
(WebCore::SubtleCrypto::exportKey):
(WebCore::SubtleCrypto::wrapKey):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmEd25519Cocoa.cpp:
(WebCore::signEd25519):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmX25519Cocoa.cpp:
(WebCore::CryptoAlgorithmX25519::platformDeriveBits):
* Source/WebCore/crypto/cocoa/CryptoKeyECMac.cpp:
(WebCore::CryptoKeyEC::platformImportPkcs8):
* Source/WebCore/css/CSSVariableData.cpp:
(WebCore::CSSVariableData::CSSVariableData):
* Source/WebCore/css/parser/CSSParserTokenRange.h:
(WebCore::CSSParserTokenRange::span const):
* Source/WebCore/dom/ElementData.cpp:
(WebCore::UniqueElementData::UniqueElementData):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::sanitizeMarkupWithArchive):
* Source/WebCore/html/FormController.cpp:
(WebCore::AtomStringVectorReader::consumeSubvector):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.cpp:
(WebCore::HTMLMetaCharsetParser::checkForMetaCharset):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.h:
* Source/WebCore/loader/FormSubmission.cpp:
(WebCore::appendMailtoPostFormDataToURL):
* Source/WebCore/loader/TextResourceDecoder.cpp:
(WebCore::bytesEqual):
(WebCore::find):
(WebCore::findTextEncoding):
(WebCore::KanjiCode::judge):
(WebCore::shouldPrependBOM):
(WebCore::TextResourceDecoder::textFromUTF8):
(WebCore::findXMLEncoding):
(WebCore::TextResourceDecoder::checkForBOM):
(WebCore::TextResourceDecoder::checkForCSSCharset):
(WebCore::TextResourceDecoder::checkForHeadCharset):
(WebCore::TextResourceDecoder::checkForMetaCharset):
(WebCore::TextResourceDecoder::detectJapaneseEncoding):
(WebCore::TextResourceDecoder::decode):
(WebCore::TextResourceDecoder::flush):
(WebCore::TextResourceDecoder::decodeAndFlush):
* Source/WebCore/loader/TextResourceDecoder.h:
(WebCore::TextResourceDecoder::decode):
(WebCore::TextResourceDecoder::decodeAndFlush):
* Source/WebCore/platform/SharedBuffer.cpp:
(WebCore::FragmentedSharedBuffer::append):
* Source/WebCore/platform/SharedBuffer.h:
(WebCore::DataSegment::bytes const):
(WebCore::FragmentedSharedBuffer::append):
* Source/WebCore/platform/cocoa/ContentFilterUnblockHandlerCocoa.mm:
(WebCore::ContentFilterUnblockHandler::webFilterEvaluatorData const):
* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp:
(WebCore::DecomposedGlyphs::create):
* Source/WebCore/platform/graphics/ImageBackingStore.h:
(WebCore::ImageBackingStore::ImageBackingStore):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::shouldWaitForLoadingOfResource):
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::cookieFromOpusCookieContents):
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs):
* Source/WebCore/platform/graphics/iso/ISOBox.cpp:
(WebCore::ISOBox::parse):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.cpp:
(WebCore::MediaRecorderPrivateMock::fetchData):
* Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm:
(WebCore::AVVideoCaptureSource::captureOutputDidFinishProcessingPhoto):
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::create):
(WebCore::FormDataElement::isolatedCopy const):
(WebCore::FormData::appendData):
(WebCore::FormData::appendMultiPartFileValue):
(WebCore::FormData::appendMultiPartStringValue):
(WebCore::FormData::appendMultiPartKeyValuePairItems):
(WebCore::FormData::appendNonMultiPartKeyValuePairItems):
(WebCore::appendBlobResolved):
(WebCore::FormData::resolveBlobReferences):
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/FormDataBuilder.cpp:
(WebCore::FormDataBuilder::append):
(WebCore::FormDataBuilder::appendQuoted):
(WebCore::FormDataBuilder::appendFormURLEncoded):
(WebCore::FormDataBuilder::beginMultiPartHeader):
(WebCore::FormDataBuilder::addBoundaryToMultiPartHeader):
(WebCore::FormDataBuilder::addFilenameToMultiPartHeader):
(WebCore::FormDataBuilder::addContentTypeToMultiPartHeader):
(WebCore::FormDataBuilder::finishMultiPartHeader):
(WebCore::FormDataBuilder::addKeyValuePairAsFormData):
(WebCore::FormDataBuilder::encodeStringAsFormData):
* Source/WebCore/platform/network/FormDataBuilder.h:
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdateResourceHTTPBody):
* Source/WebCore/platform/sql/SQLiteStatement.cpp:
(WebCore::SQLiteStatement::columnBlob):
* Source/WebCore/platform/text/cocoa/LocaleCocoa.mm:
(WebCore::LocaleCocoa::monthLabels):
(WebCore::LocaleCocoa::shortMonthLabels):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::deserializeBuffer const):
(WebCore::Internals::createPushSubscription):
* Source/WebCore/testing/ServiceWorkerInternals.cpp:
(WebCore::ServiceWorkerInternals::schedulePushEvent):
(WebCore::ServiceWorkerInternals::createPushSubscription):
* Source/WebCore/workers/service/background-fetch/BackgroundFetch.cpp:
(WebCore::BackgroundFetch::doStore):
* Source/WebCore/xml/XMLHttpRequest.cpp:
(WebCore::XMLHttpRequest::send):
(WebCore::XMLHttpRequest::sendBytesData):
* Source/WebCore/xml/XMLHttpRequest.h:
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::openFunc):
* Source/WebGPU/WebGPU/Adapter.mm:
(WebGPU::Adapter::requestDevice):
* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::Device::createBindGroup):
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(WebGPU::Device::createBindGroupLayout):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::setBindGroup):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(WebGPU::RenderBundle::RenderBundle):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::RenderBundleEncoder::RenderBundleEncoder):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::RenderPassEncoder):
(WebGPU::RenderPassEncoder::setBindGroup):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::RenderPipeline::RenderPipeline):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Device::createTexture):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
(WebKit::WebCore::vectorCopyCast):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteBuffer.cpp:
(WebKit::RemoteBuffer::mapAsync):
(WebKit::RemoteBuffer::getMappedRange):
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
(WebKit::encodeRecordHeader):
(WebKit::encodeRecordBody):
(WebKit::encodeRecord):
* Source/WebKit/Platform/IPC/cocoa/DaemonConnectionCocoa.mm:
(WebKit::Daemon::ConnectionToMachService&lt;Traits&gt;::sendWithReply const):
* Source/WebKit/Shared/API/c/WKURLRequest.cpp:
(WKURLRequestCopySettingHTTPBody):
* Source/WebKit/Shared/Cocoa/WebPushMessageCocoa.mm:
(WebKit::WebPushMessage::fromDictionary):
* Source/WebKit/Shared/RTCNetwork.cpp:
(WebKit::RTCNetwork::RTCNetwork):
(WebKit::description):
(WebKit::RTC::Network::SocketAddress::SocketAddress):
* Source/WebKit/Shared/WebPushMessage.cpp:
(WebKit::WebPushMessage::notificationPayloadToCoreData const):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(dataReferenceFrom):
* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SubFrameSOAuthorizationSession.mm:
(WebKit::SubFrameSOAuthorizationSession::fallBackToWebPathInternal):
(WebKit::SubFrameSOAuthorizationSession::completeInternal):
(WebKit::SubFrameSOAuthorizationSession::beforeStart):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/CcidConnection.mm:
(WebKit::CcidConnection::detectContactless):
(WebKit::CcidConnection::trySelectFidoApplet):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidConnection.mm:
(WebKit::reportReceived):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
(WebKit::LocalAuthenticatorInternal::aaguidVector):
* Source/WebKit/UIProcess/WebURLSchemeTask.cpp:
(WebKit::WebURLSchemeTask::didComplete):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePassEncoderProxy.cpp:
(WebKit::WebGPU::RemoteComputePassEncoderProxy::setBindGroup):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.cpp:
(WebKit::WebGPU::RemoteQueueProxy::writeBuffer):
(WebKit::WebGPU::RemoteQueueProxy::writeTexture):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderBundleEncoderProxy::setBindGroup):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderPassEncoderProxy::setBindGroup):
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
(WebKit::LibWebRTCCodecs::setEncodingConfiguration):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::firePushEvent):
* Source/WebKit/WebProcess/WebCoreSupport/SessionStateConversion.cpp:
(WebKit::toFormData):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::handleSelectionServiceClick):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):
* Source/WebKit/webpushd/ApplePushServiceConnection.mm:
(-[_WKAPSConnectionDelegate connection:didReceivePublicToken:]):
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::WebPushDaemon::getPendingPushMessages):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.cpp:
(WebCore::WebSocketChannel::send):
(WebCore::WebSocketChannel::startClosingHandshake):
(WebCore::WebSocketChannel::processFrame):
(WebCore::WebSocketChannel::enqueueRawFrame):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.h:
* Tools/TestWebKitAPI/Tests/WTF/NativePromise.cpp:
(TestWebKitAPI::PhotoProducer::takePhotoImpl const):
* Tools/TestWebKitAPI/Tests/WTF/Vector.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
(TestWebKitAPI::signUnlinkableTokenAndSendSecretToken):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
(respondToRangeRequests):

Canonical link: <a href="https://commits.webkit.org/276220@main">https://commits.webkit.org/276220@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e6878e7eb37e7cda7583d5a0630541a56317092f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44060 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46493 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46700 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40105 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46365 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27046 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20521 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36323 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44638 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20157 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37948 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/17348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39045 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2110 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/37403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40231 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39335 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48286 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/43638 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19046 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15612 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43176 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/20421 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/41901 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20636 "Hash e6878e7e for PR 25854 does not build (failure)") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/50699 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6036 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/20048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10228 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

justinmichaud reviewed (approved):
r=me for the JSC changes. I can't really comment on the WebCore stuff, but it seems reasonable. I'll let a WebCore reviewer review that part.

darinadler commented on Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
```diff
@@ -1578,7 +1578,7 @@ JSValue IntlDateTimeFormat::formatRange(JSGlobalObject* globalObject, double sta
         return { };
     }
-    Vector<UChar, 32> buffer(formattedStringPointer, formattedStringLength);
+    Vector<UChar, 32> buffer(std::span<const UChar> { formattedStringPointer, static_cast<size_t>(formattedStringLength) });
```
Can we do `std::span` without a template argument here?

cdumez replied:
Yes it works. Not sure why I added it. Fixing.

cdumez replied:
Actually, now I know why I added them. Some bots (the Igalia JSC ones) really want the explicit template here:
```
[ 59%] Building CXX object Source/JavaScriptCore/CMakeFiles/JavaScriptCore.dir/runtime/IntlPluralRules.cpp.o
/home/buildbot/igalia-jsc32-armv7-ews/JSC-ARMv7-32bits-Build-EWS/build/Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp: In member function ‘JSC::JSValue JSC::IntlDateTimeFormat::formatRange(JSC::JSGlobalObject*, double, double)’:
/home/buildbot/igalia-jsc32-armv7-ews/JSC-ARMv7-32bits-Build-EWS/build/Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:1581:30: error: ‘auto’ parameter not permitted in this context
 1581 |     Vector<UChar, 32> buffer(std::span { formattedStringPointer, static_cast<size_t>(formattedStringLength) });
      |                              ^~~
/home/buildbot/igalia-jsc32-armv7-ews/JSC-ARMv7-32bits-Build-EWS/build/Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp: In member function ‘JSC::JSValue JSC::IntlDateTimeFormat::formatRangeToParts(JSC::JSGlobalObject*, double, double)’:
/home/buildbot/igalia-jsc32-armv7-ews/JSC-ARMv7-32bits-Build-EWS/build/Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:1697:30: error: ‘auto’ parameter not permitted in this context
```

darinadler commented on Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
```diff
@@ -1694,7 +1694,7 @@ JSValue IntlDateTimeFormat::formatRangeToParts(JSGlobalObject* globalObject, dou
         return { };
     }
-    Vector<UChar, 32> buffer(formattedStringPointer, formattedStringLength);
+    Vector<UChar, 32> buffer(std::span<const UChar> { formattedStringPointer, static_cast<size_t>(formattedStringLength) });
```
Can we do `std::span` without a template argument here?

darinadler commented on Source/JavaScriptCore/wasm/WasmParser.h:
```diff
@@ -63,6 +63,7 @@ class Parser {
 
     const uint8_t* source() const { return m_source; }
     size_t length() const { return m_sourceLength; }
+    std::span<const uint8_t> sourceSpan() const { return std::span { m_source, m_sourceLength }; }
```
Can we leave out `std::span` entirely here?

darinadler commented on Source/WTF/wtf/FileSystem.cpp:
```diff
@@ -909,7 +909,7 @@ String createTemporaryDirectory()
     std::string newTempDirTemplate = tempDir + "XXXXXXXX";
 
-    Vector<char> newTempDir(newTempDirTemplate.c_str(), newTempDirTemplate.size());
+    Vector<char> newTempDir(std::span<const char> { newTempDirTemplate.data(), newTempDirTemplate.size() });
```
Seems like a helper to convert a `std::string` to a `std::span` would make sense. Doesn’t seem like an unusual operation. I won’t insist on it.

cdumez replied:
Yes, we actually convert std::string into a span several times in this patch. I'll look into adding a convenience function to StdLibExtras.h. Would `std::span<const char> toSpan(const std::string&)` sound OK?

darinadler replied:
It’s OK with me.

I'm not sure how we are deciding when to mention the names of arguments. I’d love to have a `toSpan` or `span` function overloaded to work with lots of different types. But I noticed that we have functions with names like `spanFromNSData` instead of just an overloaded function. And member functions named `span` in other cases. That’s all OK but a broader pattern might make this nicer.

cdumez replied:
Oh, maybe I should rename `spanFromNSData` to `toSpan` too then. Having overloaded functions seems good to me and `toSpan` is nice and short.

cdumez replied:
Oh, there is no `spanFromNSData`. I am guessing this was just an example. We may well need such a function though.

darinadler replied:
I confused myself. I suggested `spanFromNSData`, but it doesn’t exist yet. But same may apply for `vectorFromNSData`/`toVector`.

darinadler commented on Source/WTF/wtf/Vector.h:
```diff
@@ -803,6 +801,11 @@ class Vector : private VectorBuffer<T, inlineCapacity, Malloc> {
     bool isEmpty() const { return !size(); }
     std::span<const T> span() const { return { data(), size() }; }
 
+    Vector<T> subVector(size_t offset, size_t length = std::dynamic_extent) const
```
I think we could lowercase the V here and have the coined word be “subvector”.

darinadler commented on Source/WTF/wtf/Vector.h:
```diff
@@ -2098,6 +2101,7 @@ inline Vector<typename CopyOrMoveToVectorResult<Collection>::Type> moveToVector(
 }
 
 template<typename U> Vector(const U*, size_t) -> Vector<U>;
+template<typename U, size_t Extent> Vector(std::span<const U, Extent>) -> Vector<U>;
```
Why `U` and not `T`?

cdumez replied:
I based copied the line above. I can use `T` in both.

darinadler commented on Source/WTF/wtf/cocoa/VectorCocoa.h:
```diff
@@ -114,7 +114,7 @@ template<typename MapFunctionType> Vector<typename std::invoke_result_t<MapFunct
 
 inline Vector<uint8_t> vectorFromNSData(NSData* data)
```
Seems like we should have a `spanFromNSData` that is used here.

darinadler commented on Source/WTF/wtf/cf/VectorCF.h:
```diff
@@ -162,7 +162,7 @@ template<typename MapLambdaType> Vector<typename LambdaTypeTraits<MapLambdaType>
 
 inline Vector<uint8_t> vectorFromCFData(CFDataRef data)
```
Seems like we should have a `spanFromCFData` that is used here.

darinadler commented on Source/WTF/wtf/text/IntegerToStringConversion.h:
```diff
@@ -134,7 +135,8 @@ template<size_t N>
     using ReturnType = Vector<LChar, N>;
     using AdditionalArgumentType = void;
-    static ReturnType flush(LChar* characters, unsigned length, void*) { return { characters, length }; }
+    // FIXME: Should take in a std::span.
```
Yes!

darinadler commented on Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
```diff
@@ -138,19 +138,18 @@ FetchBodyConsumer::~FetchBodyConsumer() = default;
-        const uint8_t* bodyBegin = headerEnd + strlen("\r\n\r\n");
-        size_t bodyLength = partLength - (bodyBegin - headerBegin);
+        auto* bodyBegin = headerEnd + strlen("\r\n\r\n");
+        std::span<const uint8_t> body { bodyBegin, partLength - (bodyBegin - headerBytes.data()) };
```
Can we leave out the template argument here?

cdumez replied:
Yes, fixing.

darinadler commented on Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
```diff
@@ -516,7 +516,7 @@ static String sanitizeMarkupWithArchive(LocalFrame& frame, Document& destination
 
         CString utf8 = subframeMarkup.utf8();
-        auto blob = Blob::create(&destinationDocument, Vector { utf8.dataAsUInt8Ptr(), utf8.length() }, type);
+        auto blob = Blob::create(&destinationDocument, Vector<uint8_t> { utf8.bytes() }, type);
```
Can we add a deduction guide so this will get the type right without the template argument?

cdumez replied:
Yes, will make it work.

cdumez replied:
`Vector(utf8.bytes())` actually does work. `Vector { utf8.bytes() }` doesn't because it is trying to constructor a `Vector<std::span<const uint8_t>>`.

darinadler commented on Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
```diff
@@ -210,7 +210,7 @@ static inline NSURLRequestCachePolicy toPlatformRequestCachePolicy(ResourceReque
 {
     if (NSData* bodyData = [m_nsRequest HTTPBody])
-        m_httpBody = FormData::create([bodyData bytes], [bodyData length]);
+        m_httpBody = FormData::create(std::span { static_cast<const uint8_t*>([bodyData bytes]), [bodyData length] });
```
This is an example of where we’d use a `spanFromNSData` function.

cdumez replied:
yes, now calling `toSpan()`

darinadler commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -7219,9 +7218,9 @@ void Internals::retainTextIteratorForDocumentContent()
-    Vector<uint8_t> myAuth { static_cast<const uint8_t*>(auth.data()), auth.byteLength() };
+    auto myServerVAPIDPublicKey = serverVAPIDPublicKey.toVector();
+    auto myClientECDHPublicKey = clientECDHPublicKey.toVector();
+    auto myAuth = auth.toVector();
```
Now that these are so much less wordy, I think it would be better to just call them below and not have local variables or `WTFMove` involved.

darinadler commented on Source/WebCore/testing/ServiceWorkerInternals.cpp:
```diff
@@ -183,9 +183,9 @@ void ServiceWorkerInternals::lastNavigationWasAppInitiated(Ref<DeferredPromise>&
-    Vector<uint8_t> myAuth { static_cast<const uint8_t*>(auth.data()), auth.byteLength() };
+    auto myServerVAPIDPublicKey = serverVAPIDPublicKey.toVector();
+    auto myClientECDHPublicKey = clientECDHPublicKey.toVector();
+    auto myAuth = auth.toVector();
```
Now that these are so much less wordy, I think it would be better to just call them below and not have local variables or `WTFMove` involved.

darinadler commented on Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
```diff
@@ -86,7 +86,7 @@ static bool shouldUpdateQuery()
 const uint64_t counter = 0;
 // This aaguid is unattested.
-const uint8_t aaguid[] = { 0xFB, 0xFC, 0x30, 0x07, 0x15, 0x4E, 0x4E, 0xCC, 0x8C, 0x0B, 0x6E, 0x02, 0x05, 0x57, 0xD7, 0xBD }; // Randomly generated.
+const std::array<uint8_t, 16> aaguid = { 0xFB, 0xFC, 0x30, 0x07, 0x15, 0x4E, 0x4E, 0xCC, 0x8C, 0x0B, 0x6E, 0x02, 0x05, 0x57, 0xD7, 0xBD }; // Randomly generated.
```
Consider using `constexpr` instead of `const`.

darinadler commented on Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.cpp:
```diff
@@ -183,13 +183,13 @@ ThreadableWebSocketChannel::SendResult WebSocketChannel::send(Blob& binaryData)
 
-    LOG(Network, "WebSocketChannel %p send() Sending uint8_t* data=%p length=%d", this, data, length);
-    enqueueRawFrame(WebSocketFrame::OpCodeBinary, data, length);
+    LOG(Network, "WebSocketChannel %p send() Sending uint8_t* data=%p length=%lu", this, data.data(), data.size());
```
`%lu` isn’t right for `size_t`, should be `%zu`

darinadler commented on Tools/TestWebKitAPI/Tests/WTF/NativePromise.cpp:
```diff
@@ -1855,7 +1855,7 @@ class PhotoProducer : public ThreadSafeRefCounted<PhotoProducer> {
             // on a particular thread.
             // The consumer specifies the thread on which it wants to be called back.
-            producer.resolve(std::make_pair<Vector<uint8_t>, String>({ image.data(), image.size() }, { mimeType.data(), static_cast<unsigned>(mimeType.size()) }));
+            producer.resolve(std::make_pair<Vector<uint8_t>, String>(std::span { image.data(), image.size() }, { mimeType.data(), static_cast<unsigned>(mimeType.size()) }));
```
Surprising that we can’t make a `std::span` from a `std::vector` without separately getting `data` and `size`.

cdumez replied:
It does work. I'll fix.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
```diff
@@ -259,7 +260,7 @@ static void signUnlinkableTokenAndSendSecretToken(TokenSigningParty signingParty
         (__bridge id)kSecAttrKeyClass: (__bridge id)kSecAttrKeyClassPublic
     }, nil));
-    Vector<uint8_t> rawKeyBytes(static_cast<const uint8_t*>(publicKey.get().bytes), publicKey.get().length);
+    Vector<uint8_t> rawKeyBytes = vectorFromNSData(publicKey.get());
```
Should get rid of the local variable and `WTFMove` here since this is so short now.

darinadler reviewed (approved):
It’s great to move more to `uint8_t` instead of `char`, but I can’t help thinking we might want to use `std::byte` some day.

Seems like we should add more helper functions to make spans so there aren’t so many `static_cast<size_t>` and `reinterpret_cast` throughout the code. There seem to be repeated patterns like converting C strings to spans that aren’t using functions.

To state the obvious, and what I know you are already intending, we want to convert pointer and length pairs into spans as close to the place they cross over into WebKit as possible.

szewai commented on Source/WebCore/Modules/async-clipboard/ClipboardItem.cpp:
```diff
@@ -42,7 +42,7 @@ ClipboardItem::~ClipboardItem() = default;
 {
     auto utf8 = stringData.utf8();
-    return Blob::create(context, Vector { utf8.dataAsUInt8Ptr(), utf8.length() }, Blob::normalizedContentType(type));
+    return Blob::create(context, Vector(utf8.bytes()), Blob::normalizedContentType(type));
```
`return Blob::create(context, Vector(stringData.utf8().bytes()), Blob::normalizedContentType(type));`

szewai commented on Source/WebCore/Modules/push-api/PushEvent.cpp:
```diff
@@ -41,14 +41,14 @@ static Vector<uint8_t> dataFromPushMessageDataInit(PushMessageDataInit& data)
     }, [](String& value) -> Vector<uint8_t> {
         auto utf8 = value.utf8();
-        return Vector<uint8_t> { reinterpret_cast<const uint8_t*>(utf8.data()), utf8.length() };
+        return { utf8.bytes() };
```
`return { value.utf8().bytes() };`

szewai commented on Source/WebCore/Modules/push-api/PushEvent.cpp:
```diff
@@ -41,14 +41,14 @@ static Vector<uint8_t> dataFromPushMessageDataInit(PushMessageDataInit& data)
         if (!value)
             return { };
-        return { reinterpret_cast<const uint8_t*>(value->data()), value->byteLength() };
+        return { value->bytes() };
```
`return value ? Vector<uint8_t> { value->bytes() } : Vector<uint8_t> { };`

darinadler replied:
I think we should use a helper function instead of repeating this idiom.

cdumez replied:
That's kind of verbose, I think I prefer to use:
```
if (!value)
     return { };
return value->bytes();
```

szewai commented on Source/WebCore/Modules/push-api/PushEvent.cpp:
```diff
@@ -41,14 +41,14 @@ static Vector<uint8_t> dataFromPushMessageDataInit(PushMessageDataInit& data)
         if (!value)
             return { };
-        return { reinterpret_cast<const uint8_t*>(value->baseAddress()), value->byteLength() };
+        return { value->bytes() };
```
`return value ? Vector<uint8_t> { value->bytes() } : Vector<uint8_t> { };`

cdumez replied:
ditto.

szewai commented on Source/WebCore/Modules/push-api/PushManager.cpp:
```diff
@@ -88,11 +88,11 @@ void PushManager::subscribe(ScriptExecutionContext& context, std::optional<PushS
             if (!value)
                 return Vector<uint8_t> { };
-            return Vector<uint8_t> { reinterpret_cast<const uint8_t*>(value->data()), value->byteLength() };
+            return Vector<uint8_t>(value->bytes());
```
`return value ? Vector<uint8_t> { value->bytes() } : Vector<uint8_t> { };`

szewai commented on Source/WebCore/Modules/push-api/PushManager.cpp:
```diff
@@ -88,11 +88,11 @@ void PushManager::subscribe(ScriptExecutionContext& context, std::optional<PushS
             if (!value)
                 return Vector<uint8_t> { };
-            return Vector<uint8_t> { reinterpret_cast<const uint8_t*>(value->baseAddress()), value->byteLength() };
+            return Vector<uint8_t>(value->bytes());
```
`return value ? Vector<uint8_t> { value->bytes() } : Vector<uint8_t> { };`

szewai commented on Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
```diff
@@ -516,7 +516,7 @@ static String sanitizeMarkupWithArchive(LocalFrame& frame, Document& destination
 
         CString utf8 = subframeMarkup.utf8();
-        auto blob = Blob::create(&destinationDocument, Vector { utf8.dataAsUInt8Ptr(), utf8.length() }, type);
+        auto blob = Blob::create(&destinationDocument, Vector<uint8_t> { utf8.bytes() }, type);
```
`auto blob = Blob::create(&destinationDocument, Vector<uint8_t> { subframeMarkup.utf8().bytes() }, type);`

szewai commented on Source/WebCore/loader/FormSubmission.cpp:
```diff
@@ -74,8 +74,8 @@ static void appendMailtoPostFormDataToURL(URL& url, const FormData& data, const
-    Vector<char> bodyData;
-    bodyData.append("body=", 5);
+    Vector<uint8_t> bodyData;
+    bodyData.append(std::span { reinterpret_cast<const uint8_t*>("body="), 5 });
```
`Vector<uint8_t> bodyData { std::span { reinterpret_cast<const uint8_t*>("body="), 5 } };`

cdumez replied:
Looks like I can go one step further and use `Vector<uint8_t> bodyData(std::span { "body=", 5 });`. No more `reinterpret_cast`.

szewai commented on Source/WebCore/platform/audio/gstreamer/PlatformRawAudioDataGStreamer.cpp:
```diff
@@ -89,7 +89,7 @@ RefPtr<PlatformRawAudioData> PlatformRawAudioData::create(std::span<const uint8_
 
-    Vector<uint8_t> dataStorage { sourceData.data(), sourceData.size() };
+    Vector<uint8_t> dataStorage { sourceData };
     auto data = SharedBuffer::create(WTFMove(dataStorage));
```
`auto data = SharedBuffer::create(Vector<uint8_t> { sourceData });`

szewai commented on Source/WebCore/platform/network/FormData.cpp:
```diff
@@ -322,7 +322,7 @@ static void appendBlobResolved(BlobRegistryImpl* blobRegistry, FormData& formDat
         if (blobItem.type() == BlobDataItem::Type::Data) {
             ASSERT(blobItem.data());
-            formData.appendData(blobItem.data()->data() + static_cast<int>(blobItem.offset()), static_cast<int>(blobItem.length()));
+            formData.appendData(blobItem.data()->bytes().subspan(blobItem.offset(), blobItem.length()));
```
Should `appendData` just take a std::span?

cdumez replied:
It does take a span and I am passing it a span.

szewai commented on Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.cpp:
```diff
@@ -81,7 +81,7 @@ void RemoteQueueProxy::writeBuffer(
         return;
 
-    auto sendResult = send(Messages::RemoteQueue::WriteBuffer(convertedBuffer, bufferOffset, Vector<uint8_t>(static_cast<const uint8_t*>(source) + dataOffset, size.value_or(byteLength - dataOffset))));
+    auto sendResult = send(Messages::RemoteQueue::WriteBuffer(convertedBuffer, bufferOffset, Vector(std::span { static_cast<const uint8_t*>(source), byteLength }.subspan(dataOffset))));
```
Is `Vector` needed?

cdumez replied:
It doesn't appear so. Will drop.

mwyrzykowski commented on Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.cpp:
```diff
@@ -81,7 +81,7 @@ void RemoteQueueProxy::writeBuffer(
     if (!convertedBuffer)
         return;
 
-    auto sendResult = send(Messages::RemoteQueue::WriteBuffer(convertedBuffer, bufferOffset, Vector<uint8_t>(static_cast<const uint8_t*>(source) + dataOffset, size.value_or(byteLength - dataOffset))));
```
I don't think we can ignore the `size` argument, can you please correct before merging?

cdumez replied:
It looks like I did inadvertently drop `size` in the new code. I'll fix that. Thanks for catching.

mwyrzykowski reviewed (changes requested):
I think the `size` member is dropped in RemoteQueueProxy and that is likely to cause a regression unless I am misunderstanding something

cdumez committed:
Drop Vector constructor taking a raw pointer and a size
https://bugs.webkit.org/show_bug.cgi?id=270944

Reviewed by Sihui Liu, Mike Wyrzykowski, Justin Michaud and Darin Adler.

Drop Vector constructor taking a raw pointer and a size and port call sites to
the constructor taking a std::span instead.

This is work towards more widespread std::span adoption in WebKit, which has
security benefits.

* Source/JavaScriptCore/runtime/ArrayBuffer.cpp:
(JSC::ArrayBuffer::tryCreate):
* Source/JavaScriptCore/runtime/ArrayBuffer.h:
* Source/JavaScriptCore/runtime/ArrayBufferView.h:
(JSC::ArrayBufferView::bytes const):
* Source/JavaScriptCore/runtime/ISO8601.cpp:
(JSC::ISO8601::parseTimeZoneBracketedAnnotation):
(JSC::ISO8601::parseCalendar):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::formatRange):
(JSC::IntlDateTimeFormat::formatRangeToParts):
* Source/JavaScriptCore/wasm/WasmParser.h:
(JSC::Wasm::Parser::sourceSpan const):
* Source/JavaScriptCore/wasm/WasmSectionParser.cpp:
(JSC::Wasm::SectionParser::parseInitExpr):
* Source/JavaScriptCore/wasm/WasmStreamingParser.cpp:
(JSC::Wasm::StreamingParser::consume):
* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
(PropertyData.convertStringToCppFormat):
* Source/WTF/wtf/Algorithms.h:
(WTF::equalSpans):
* Source/WTF/wtf/Vector.h:
(WTF::Vector::Vector):
(WTF::Vector::subVector const):
* Source/WTF/wtf/cf/VectorCF.h:
(WTF::vectorFromCFData):
* Source/WTF/wtf/cocoa/VectorCocoa.h:
(WTF::vectorFromNSData):
* Source/WTF/wtf/persistence/PersistentEncoder.h:
(WTF::Persistence::Encoder::bytes const):
* Source/WTF/wtf/text/IntegerToStringConversion.h:
* Source/WTF/wtf/text/StringParsingBuffer.h:
* Source/WTF/wtf/text/WTFString.cpp:
(asciiDebug):
* Source/WTF/wtf/text/WTFString.h:
(WTF::String::fromUTF8):
* Source/WebCore/Modules/ShapeDetection/Implementation/Cocoa/FaceDetectorImplementation.mm:
(WebCore::ShapeDetection::convertLandmark):
* Source/WebCore/Modules/async-clipboard/ClipboardItem.cpp:
(WebCore::ClipboardItem::blobFromString):
* Source/WebCore/Modules/async-clipboard/ios/ClipboardImageReaderIOS.mm:
(WebCore::ClipboardImageReader::readBuffer):
* Source/WebCore/Modules/async-clipboard/mac/ClipboardImageReaderMac.mm:
(WebCore::ClipboardImageReader::readBuffer):
* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::consumeArrayBuffer):
(WebCore::FetchBody::consumeArrayBufferView):
(WebCore::FetchBody::consumeText):
(WebCore::FetchBody::bodyAsFormData const):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::FetchBodyConsumer::packageFormData):
(WebCore::resolveWithTypeAndData):
(WebCore::FetchBodyConsumer::resolveWithData):
(WebCore::FetchBodyConsumer::resolveWithFormData):
(WebCore::FetchBodyConsumer::resolve):
(WebCore::FetchBodyConsumer::takeAsText):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.h:
* Source/WebCore/Modules/fetch/FetchBodyOwner.cpp:
(WebCore::FetchBodyOwner::arrayBuffer):
(WebCore::FetchBodyOwner::formData):
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::RTCDataChannel::didReceiveRawData):
* Source/WebCore/Modules/notifications/NotificationDataCocoa.mm:
(WebCore::NotificationData::fromDictionary):
* Source/WebCore/Modules/push-api/PushEvent.cpp:
(WebCore::dataFromPushMessageDataInit):
* Source/WebCore/Modules/push-api/PushManager.cpp:
(WebCore::PushManager::subscribe):
* Source/WebCore/Modules/push-api/PushMessageCrypto.cpp:
(WebCore::PushCrypto::ClientKeys::generate):
(WebCore::PushCrypto::decryptAESGCMPayload):
* Source/WebCore/Modules/push-api/PushMessageData.cpp:
(WebCore::PushMessageData::text):
* Source/WebCore/Modules/push-api/cocoa/PushCryptoCocoa.cpp:
(WebCore::PushCrypto::P256DHKeyPair::generate):
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp:
(WebCore::convertBytesToVector):
* Source/WebCore/Modules/webauthn/cbor/CBORValue.cpp:
(cbor::CBORValue::CBORValue):
* Source/WebCore/Modules/webauthn/fido/DeviceResponseConverter.cpp:
(fido::decodeResponseMap):
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.cpp:
(fido::FidoHidInitPacket::createFromSerializedData):
(fido::FidoHidContinuationPacket::createFromSerializedData):
* Source/WebCore/Modules/webauthn/fido/Pin.cpp:
(fido::pin::encodeCOSEPublicKey):
* Source/WebCore/Modules/webauthn/fido/U2fResponseConverter.cpp:
(fido::WebCore::extractECPublicKeyFromU2fRegistrationResponse):
(fido::WebCore::extractCredentialIdFromU2fRegistrationResponse):
(fido::WebCore::createFidoAttestationStatementFromU2fRegisterResponse):
(fido::readU2fSignResponse):
* Source/WebCore/PAL/pal/text/TextCodec.cpp:
(PAL::TextCodec::decode):
* Source/WebCore/PAL/pal/text/TextCodec.h:
* Source/WebCore/PAL/pal/text/TextEncodingDetector.h:
* Source/WebCore/PAL/pal/text/TextEncodingDetectorICU.cpp:
(PAL::detectTextEncoding):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
(WebCore::fulfillPromiseWithArrayBuffer):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::toKeyData):
(WebCore::copyToVector):
(WebCore::SubtleCrypto::encrypt):
(WebCore::SubtleCrypto::decrypt):
(WebCore::SubtleCrypto::sign):
(WebCore::SubtleCrypto::digest):
(WebCore::SubtleCrypto::deriveBits):
(WebCore::SubtleCrypto::exportKey):
(WebCore::SubtleCrypto::wrapKey):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmEd25519Cocoa.cpp:
(WebCore::signEd25519):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmX25519Cocoa.cpp:
(WebCore::CryptoAlgorithmX25519::platformDeriveBits):
* Source/WebCore/crypto/cocoa/CryptoKeyECMac.cpp:
(WebCore::CryptoKeyEC::platformImportPkcs8):
* Source/WebCore/css/CSSVariableData.cpp:
(WebCore::CSSVariableData::CSSVariableData):
* Source/WebCore/css/parser/CSSParserTokenRange.h:
(WebCore::CSSParserTokenRange::span const):
* Source/WebCore/dom/ElementData.cpp:
(WebCore::UniqueElementData::UniqueElementData):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::sanitizeMarkupWithArchive):
* Source/WebCore/html/FormController.cpp:
(WebCore::AtomStringVectorReader::consumeSubvector):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.cpp:
(WebCore::HTMLMetaCharsetParser::checkForMetaCharset):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.h:
* Source/WebCore/loader/FormSubmission.cpp:
(WebCore::appendMailtoPostFormDataToURL):
* Source/WebCore/loader/TextResourceDecoder.cpp:
(WebCore::bytesEqual):
(WebCore::find):
(WebCore::findTextEncoding):
(WebCore::KanjiCode::judge):
(WebCore::shouldPrependBOM):
(WebCore::TextResourceDecoder::textFromUTF8):
(WebCore::findXMLEncoding):
(WebCore::TextResourceDecoder::checkForBOM):
(WebCore::TextResourceDecoder::checkForCSSCharset):
(WebCore::TextResourceDecoder::checkForHeadCharset):
(WebCore::TextResourceDecoder::checkForMetaCharset):
(WebCore::TextResourceDecoder::detectJapaneseEncoding):
(WebCore::TextResourceDecoder::decode):
(WebCore::TextResourceDecoder::flush):
(WebCore::TextResourceDecoder::decodeAndFlush):
* Source/WebCore/loader/TextResourceDecoder.h:
(WebCore::TextResourceDecoder::decode):
(WebCore::TextResourceDecoder::decodeAndFlush):
* Source/WebCore/platform/SharedBuffer.cpp:
(WebCore::FragmentedSharedBuffer::append):
* Source/WebCore/platform/SharedBuffer.h:
(WebCore::DataSegment::bytes const):
(WebCore::FragmentedSharedBuffer::append):
* Source/WebCore/platform/cocoa/ContentFilterUnblockHandlerCocoa.mm:
(WebCore::ContentFilterUnblockHandler::webFilterEvaluatorData const):
* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp:
(WebCore::DecomposedGlyphs::create):
* Source/WebCore/platform/graphics/ImageBackingStore.h:
(WebCore::ImageBackingStore::ImageBackingStore):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::shouldWaitForLoadingOfResource):
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::cookieFromOpusCookieContents):
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs):
* Source/WebCore/platform/graphics/iso/ISOBox.cpp:
(WebCore::ISOBox::parse):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.cpp:
(WebCore::MediaRecorderPrivateMock::fetchData):
* Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm:
(WebCore::AVVideoCaptureSource::captureOutputDidFinishProcessingPhoto):
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::create):
(WebCore::FormDataElement::isolatedCopy const):
(WebCore::FormData::appendData):
(WebCore::FormData::appendMultiPartFileValue):
(WebCore::FormData::appendMultiPartStringValue):
(WebCore::FormData::appendMultiPartKeyValuePairItems):
(WebCore::FormData::appendNonMultiPartKeyValuePairItems):
(WebCore::appendBlobResolved):
(WebCore::FormData::resolveBlobReferences):
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/FormDataBuilder.cpp:
(WebCore::FormDataBuilder::append):
(WebCore::FormDataBuilder::appendQuoted):
(WebCore::FormDataBuilder::appendFormURLEncoded):
(WebCore::FormDataBuilder::beginMultiPartHeader):
(WebCore::FormDataBuilder::addBoundaryToMultiPartHeader):
(WebCore::FormDataBuilder::addFilenameToMultiPartHeader):
(WebCore::FormDataBuilder::addContentTypeToMultiPartHeader):
(WebCore::FormDataBuilder::finishMultiPartHeader):
(WebCore::FormDataBuilder::addKeyValuePairAsFormData):
(WebCore::FormDataBuilder::encodeStringAsFormData):
* Source/WebCore/platform/network/FormDataBuilder.h:
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdateResourceHTTPBody):
* Source/WebCore/platform/sql/SQLiteStatement.cpp:
(WebCore::SQLiteStatement::columnBlob):
* Source/WebCore/platform/text/cocoa/LocaleCocoa.mm:
(WebCore::LocaleCocoa::monthLabels):
(WebCore::LocaleCocoa::shortMonthLabels):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::deserializeBuffer const):
(WebCore::Internals::createPushSubscription):
* Source/WebCore/testing/ServiceWorkerInternals.cpp:
(WebCore::ServiceWorkerInternals::schedulePushEvent):
(WebCore::ServiceWorkerInternals::createPushSubscription):
* Source/WebCore/workers/service/background-fetch/BackgroundFetch.cpp:
(WebCore::BackgroundFetch::doStore):
* Source/WebCore/xml/XMLHttpRequest.cpp:
(WebCore::XMLHttpRequest::send):
(WebCore::XMLHttpRequest::sendBytesData):
* Source/WebCore/xml/XMLHttpRequest.h:
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::openFunc):
* Source/WebGPU/WebGPU/Adapter.mm:
(WebGPU::Adapter::requestDevice):
* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::Device::createBindGroup):
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(WebGPU::Device::createBindGroupLayout):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::setBindGroup):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(WebGPU::RenderBundle::RenderBundle):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::RenderBundleEncoder::RenderBundleEncoder):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::RenderPassEncoder):
(WebGPU::RenderPassEncoder::setBindGroup):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::RenderPipeline::RenderPipeline):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Device::createTexture):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
(WebKit::WebCore::vectorCopyCast):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteBuffer.cpp:
(WebKit::RemoteBuffer::mapAsync):
(WebKit::RemoteBuffer::getMappedRange):
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
(WebKit::encodeRecordHeader):
(WebKit::encodeRecordBody):
(WebKit::encodeRecord):
* Source/WebKit/Platform/IPC/cocoa/DaemonConnectionCocoa.mm:
(WebKit::Daemon::ConnectionToMachService<Traits>::sendWithReply const):
* Source/WebKit/Shared/API/c/WKURLRequest.cpp:
(WKURLRequestCopySettingHTTPBody):
* Source/WebKit/Shared/Cocoa/WebPushMessageCocoa.mm:
(WebKit::WebPushMessage::fromDictionary):
* Source/WebKit/Shared/RTCNetwork.cpp:
(WebKit::RTCNetwork::RTCNetwork):
(WebKit::description):
(WebKit::RTC::Network::SocketAddress::SocketAddress):
* Source/WebKit/Shared/WebPushMessage.cpp:
(WebKit::WebPushMessage::notificationPayloadToCoreData const):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(dataReferenceFrom):
* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SubFrameSOAuthorizationSession.mm:
(WebKit::SubFrameSOAuthorizationSession::fallBackToWebPathInternal):
(WebKit::SubFrameSOAuthorizationSession::completeInternal):
(WebKit::SubFrameSOAuthorizationSession::beforeStart):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/CcidConnection.mm:
(WebKit::CcidConnection::detectContactless):
(WebKit::CcidConnection::trySelectFidoApplet):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidConnection.mm:
(WebKit::reportReceived):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
(WebKit::LocalAuthenticatorInternal::aaguidVector):
* Source/WebKit/UIProcess/WebURLSchemeTask.cpp:
(WebKit::WebURLSchemeTask::didComplete):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePassEncoderProxy.cpp:
(WebKit::WebGPU::RemoteComputePassEncoderProxy::setBindGroup):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.cpp:
(WebKit::WebGPU::RemoteQueueProxy::writeBuffer):
(WebKit::WebGPU::RemoteQueueProxy::writeTexture):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderBundleEncoderProxy::setBindGroup):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderPassEncoderProxy::setBindGroup):
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
(WebKit::LibWebRTCCodecs::setEncodingConfiguration):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::firePushEvent):
* Source/WebKit/WebProcess/WebCoreSupport/SessionStateConversion.cpp:
(WebKit::toFormData):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::handleSelectionServiceClick):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):
* Source/WebKit/webpushd/ApplePushServiceConnection.mm:
(-[_WKAPSConnectionDelegate connection:didReceivePublicToken:]):
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::WebPushDaemon::getPendingPushMessages):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.cpp:
(WebCore::WebSocketChannel::send):
(WebCore::WebSocketChannel::startClosingHandshake):
(WebCore::WebSocketChannel::processFrame):
(WebCore::WebSocketChannel::enqueueRawFrame):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.h:
* Tools/TestWebKitAPI/Tests/WTF/NativePromise.cpp:
(TestWebKitAPI::PhotoProducer::takePhotoImpl const):
* Tools/TestWebKitAPI/Tests/WTF/Vector.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
(TestWebKitAPI::signUnlinkableTokenAndSendSecretToken):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
(respondToRangeRequests):

Canonical link: https://commits.webkit.org/276220@main
